Script action for repairing destroyed bridges

FS-21 opened the pull request:
- Picks a Bridge Repair Hut from the map that is linked with a bridge with destroyed sections and is reachable by engineers and then send the Taskforce against it.
- The Taskforce must have any engineer and (Bridge Repair Huts in the map) or the script will end.
- Puts non-engineers into Area Guard mode when they arrive near the Bridge Repair Hut location.
- Only are evaluated Bridge Repair Huts that can be reached by the Engineers.

In aimd.ini:
[SOMESCRIPTTYPE]  ; ScriptType
x=10105,n                 ; integer, mode for selecting Bridge Repair Huts

- The possible argument values are:
0 -> Pick the closest
1 -> Pick the Farthest
-1 -> Pick Random

FS-21 committed:
initial commit

FS-21 committed:
Point to the desired YRpp

With the Bridge validation check

FS-21 committed:
Improvements and additions

Now the script action argument can be used to select a mode:
0: Closest Repair Hut with destroyed bridge.
1: Farthest Repair Hut with destroyed bridge.
-1: Random Repair Hut with destroyed bridge.

Important: now Engineers will pick only Repair Huts that can be reached.

FS-21 committed:
Point to latest YRpp with the required change

FS-21 committed:
Script action renaming and docs

Now is the action 10104 and added documentation.

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/script-action-repair-destroyed-bridge

Belonit committed:
Fix trailing whitespace

chaserli commented on src/Ext/Script/Body.cpp:
```diff
@@ -3088,3 +3092,465 @@ void ScriptExt::Stop_ForceJump_Countdown(TeamClass *pTeam)
+	// Save all Bridge Repair Hut Structures for saving time in the team actions. The worst case-scenario is if the map doesn't have bridges
+	if (pTeamData->MapPath_BridgeRepairHuts.Count == 0)
+	{
+		for (auto pTechno : *TechnoClass::Array)
+		{
+			if (pTechno->WhatAmI() == AbstractType::Building)
+			{
+				auto pBuilding = static_cast<BuildingClass*>(pTechno);
+				if (!pBuilding)
+					continue;
+
```
```suggestion
		for (auto pBuilding : *BuildingClass::Array)
		{

```
Ahhh

FS-21 committed:
YRpp

FS-21 committed:
Code rewrite

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Update docs

Script action number moved from 10104 to 10105.

FS-21 committed:
fixes & tweaks

FS-21 commented:
Internal rewrite of the feature.

Thanks to Starkku I don't need to do a pathfinding routine, simplifying the code doing the same.

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Script Action moved into its own file

It is big enough.

Also removed the struct MapPathCellElement because it was from an older obsolete feature.

FS-21 committed:
fix mistakes

FS-21 committed:
Tweak

FS-21 committed:
Fix previous commit

FS-21 committed:
Fix crash

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Small tweaks and FA2 documentation

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

FS-21 committed:
Merge branch 'develop' into feature/script-action-repair-destroyed-bridge

DeathFishAtEase committed:
[doc] Move the Whats-New entries to the correct positions and link them to the corresponding locations in the document

DeathFishAtEase committed:
Merge remote-tracking branch 'Phobos-developers/develop' into feature/script-action-repair-destroyed-bridge
